controllers/articles.controller.js

```diff
@@ -3,16 +3,20 @@ const {
 	selectArticleById,
 	updatedVotesByArticleById,
 	selectCommentsForArticleId,
-	checkArticleExists,
 	insertCommentOnArticleId
 } = require('../models/articles.model');
 
-const { checkUserExists } = require('../models/users.model');
+const { checkArticleExists, checkTopic, checkUserExists } = require('../db/seeds/utils');
 
 exports.getArticles = (req, res, next) => {
-	selectArticles()
+	const { sort_by, order, topic } = req.query;
+	const promiseArray = [ selectArticles(sort_by, order, topic) ];
+	if (topic !== undefined) {
+		promiseArray.push(checkTopic(topic));
+	}
+	Promise.all(promiseArray)
```

Comment:
it's not clear to my why Promise.all here is used instead of return. Consider a refactor when time permits